src/viem/saga.ts

```diff
@@ -1,35 +1,238 @@
 import BigNumber from 'bignumber.js'
-import erc20 from 'src/abis/IERC20.json'
-import { tokenAmountInSmallestUnit } from 'src/tokens/saga'
+import erc20 from 'src/abis/IERC20'
+import stableToken from 'src/abis/StableToken'
+import { FeeInfo } from 'src/fees/saga'
+import { encryptComment } from 'src/identity/commentEncryption'
+import { buildSendTx } from 'src/send/saga'
+import { getTokenInfo, tokenAmountInSmallestUnit } from 'src/tokens/saga'
+import { isStablecoin } from 'src/tokens/utils'
+import { chooseTxFeeDetails } from 'src/transactions/send'
+import { TransactionContext } from 'src/transactions/types'
 import Logger from 'src/utils/Logger'
 import { publicClient } from 'src/viem'
+import { ViemWallet } from 'src/viem/getLockableWallet'
 import { getViemWallet } from 'src/web3/contracts'
 import networkConfig from 'src/web3/networkConfig'
 import { call } from 'typed-redux-saga'
-import { getAddress } from 'viem'
-
-// example usage
-// todo(ACT-786): build on this when implementing viem version of sendAndMonitorTransaction
-export function* send(
-  recipientAddress: string,
-  amount: BigNumber,
-  tokenAddress: string,
+import { SimulateContractReturnType, getAddress } from 'viem'
+
+const TAG = 'viem/saga'
+
+/**
+ * Send a payment with viem. The equivalent of buildAndSendPayment in src/send/saga.
+ *
+ * @param options an object containing the arguments
+ * @param options.context the transaction context
+ * @param options.recipientAddress the address to send the payment to
+ * @param options.amount the crypto amount to send
+ * @param options.tokenAddress the crypto token address
+ * @param options.comment the comment on the transaction
+ * @param options.feeInfo an object containing the fee information
+ * @returns
+ */
+export function* sendPayment({
+  context,
+  recipientAddress,
+  amount,
+  tokenAddress,
+  comment,
+  feeInfo,
+}: {
+  context: TransactionContext
+  recipientAddress: string
+  amount: BigNumber
+  tokenAddress: string
   comment: string
-) {
+  feeInfo: FeeInfo
+}) {
   const wallet = yield* call(getViemWallet, networkConfig.viemChain.celo)
```

Comment:
Don't we need to call wallet.unlockAccount as well? If its not the first time that `getViemWallet` is being called in the app then the wallet will not necessarily be unlocked

Reply:
We currently unlock the contract kit wallet at the start of the send flow [here](https://github.com/valora-inc/wallet/blob/main/src/send/saga.ts#L312) and since locks are shared, an explicit unlock call on this wallet wasn't required. We can still call the unlock method on the viem wallet here but we'd probably need a wrapper similar [to this](https://github.com/valora-inc/wallet/blob/main/src/web3/saga.ts#L192) to get a password and then call unlock on the wallet. What do you think?

Reply:
I see, you're right then its unnecessary for now, but I still think we should do it. That code in sendPaymentSaga could change as we get off contractkit and we may be unaware of the side effects, this saga could also start being called somewhere else that doesn't unlock the account first. I think we'll have to create a wrapper unlock functioin like that eventually anyways and it would be prudent to do it here and now.

Reply:
@jophish I don't think its straightforward to update KeychainLock.unlock to not require a passphrase, I see examples where we pass different passphrases for different accounts with different passphrases in the same wallet. 
https://github.com/valora-inc/wallet/blob/4bfeab16689787c1fcc56c6c68d2f2dde02ff324/src/merchantPayment/hooks.ts#L43-L44
I guess we don't currently support this on ViemWallet, which has a 1:1 mapping between wallets and account instances which is why I believe the `unlockAccount` on ViemWallet just takes a passphrase and a duration and not the account address like the Celo wallet. I guess one option could be to have the ViemWallet's unlockAccount method to get the password and then pass it on but that seems like its something the Wallet class doesn't need to know about.